data/json/mapgen/basecamps/base/modular_hub/version_2/modular_field_common.json

```diff
@@ -0,0 +1,367 @@
+[
+  {
+    "type": "palette",
+    "id": "fbmh_2_palette_common",
+    "terrain": { ",": "t_dirt", ".": "t_region_groundcover", ";": "t_dirt" },
+    "furniture": { ";": "f_bulletin" }
+  },
+  {
+    "type": "mapgen",
+    "update_mapgen_id": "fbmh_2",
+    "method": "json",
+    "object": {
+      "set": [
```

Comment:
doing this through `set` seems like a very complex way to do it. Is there a reason you didn't just use `rows`?

Reply:
It was a fair while ago, but I remember having trouble getting it to work specifically for the initial "outline". I tried various methods before ending up copying what some old primitive implementation managed to get to work.

Reply:
I am not sure if there's some reason for it in the base expansions code, so I was mostly curious.

Reply:
Regarding the question of materials supported. No all the materials supported by the previous base camp at the are supported, plus one more. The reason I prefer to use mi-go resin for testing is that it requires the spawning of the least number of materials and tools and construction is the fastest (adding just a few days rather than weeks to the timer in debug mode). However, I intended to perform regression testing with a different material for each expansion (to repeat, I did go through each material when the work was done initially, and I'd rather not repeat it again just because it's split up and a lot of time has passed).

The structure is such that the constructions are strictly separated from the furniture so that changes to furniture don't have any effect on the (unfortunately) much more numerous materials, and material changes are just search/replace changes (with a few different variations due to how the materials are named.

Originally all variants were tested (as I believe is mentioned above), and the process was to first test/fix/test/fix... the first one, then search/replace copies of the files, test that version, and so on.

But yes, I can try to create a bazillion PRs. However, I expect the incompetent merging logic to be unable to figure out in which order to add each material in the files where they're going to be listed one after the other, resulting in conflicts that have to be resolved manually as each PR is included.